### PR TITLE
Adsk Contrib - Various adjustments to the Windows DLL

### PR DESCRIPTION
**Pull request by hodoulp**
The pull request includes:
1. Any Windows DLL must have some extra information embedded in the library binary. Waiting the completion of the ASWF transition, some are still missing. But the mechanism is now in place.
2. To avoid the infamous DLL hell on Windows change the binary file name to mimic the Linux good practice.  Refer to #605.
3. Add the WIN32_LEAN_AND_MEAN define to strip 'not often used' Windows libraries which is defined as a good practice. However, I double-checked the binary size without noticing any size change.

### CODE REVIEW

**Comment by hodoulp** [comment]
The pull request is the continuation of #596.

File: src/OpenColorIO/CMakeLists.txt [hunk 1]
```diff
@@ -143,13 +156,29 @@ if(OCIO_USE_SSE)
...
+        PRIVATE
+            XML_STATIC
+    )
+
+    if(BUILD_SHARED_LIBS)
+
+        # Impose a versioned name on Windows to avoid binary name clashes 
+        # between different versions of the same library.
+
+        # Note: Do not append the PATCH to allow Apps use the
```
Comment (michdolan) [inline src/OpenColorIO/CMakeLists.txt]:
```suggestion
        # Note: Do not append the PATCH to allow Apps to use the
```

Comment (hodoulp) [reply src/OpenColorIO/CMakeLists.txt]: Fixed

**Comment by hodoulp** [comment]
This PR qualifies for single approver rule and could be merged next week unless there is feedback.

**Comment by hodoulp** [comment]
24 hour merge warning. Please have a look if you haven't yet.

**Pull request merged**